class.ext_update.php

```diff
@@ -375,58 +452,27 @@ protected function doSolariumSolrUpdate()
+     */
+    protected function updateDocumentAddFormat(): void
     {
-        // Check if column "document_format" exists.
-        $database = $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default']['dbname'];
-        $result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
-            'COLUMN_NAME',
-            'INFORMATION_SCHEMA.COLUMNS',
-            'TABLE_NAME="tx_dlf_documents" AND TABLE_SCHEMA="' . $database . '" AND COLUMN_NAME="document_format"'
-        );
-        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
-            if ($resArray['COLUMN_NAME'] == 'document_format') {
-                if ($checkStructureOnly) {
-                    return false;
-                }
-                // Check if column has empty fields.
-                $result2 = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
-                    'uid',
-                    'tx_dlf_documents',
-                    'document_format="" OR document_format IS NULL'
-                );
-                if ($GLOBALS['TYPO3_DB']->sql_num_rows($result2) == 0) {
-                    return false;
-                }
-            }
-        }
-        return true;
-    }
 
-    protected function updateDocumentAddFormat()
-    {
-        if ($this->hasNoFormatForDocument(true)) {
-            $sqlQuery = 'ALTER TABLE tx_dlf_documents ADD COLUMN document_format varchar(100) DEFAULT "" NOT NULL;';
-            $result = $GLOBALS['TYPO3_DB']->sql_query($sqlQuery);
-            if ($result) {
-                Helper::addMessage(
-                    $GLOBALS['LANG']->getLL('update.documentAddFormatOkay', true),
-                    $GLOBALS['LANG']->getLL('update.documentAddFormat', true),
-                    \TYPO3\CMS\Core\Messaging\FlashMessage::OK
-                );
-                $this->content .= Helper::renderFlashMessages();
-            } else {
-                Helper::addMessage(
-                    $GLOBALS['LANG']->getLL('update.documentAddFormatNotOkay', true),
-                    $GLOBALS['LANG']->getLL('update.documentAddFormat', true),
-                    \TYPO3\CMS\Core\Messaging\FlashMessage::WARNING
-                );
-                $this->content .= Helper::renderFlashMessages();
-                return;
-            }
-        }
-        $sqlQuery = 'UPDATE `tx_dlf_documents` SET `document_format`="METS" WHERE `document_format` IS NULL OR `document_format`="";';
-        $result = $GLOBALS['TYPO3_DB']->sql_query($sqlQuery);
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_documents');
+
+        $result = $queryBuilder
+            ->update('tx_dlf_documents')
+            ->where(
+                $queryBuilder->expr()->orX(
+                    $queryBuilder->expr()->eq('document_format', $queryBuilder->createNamedParameter(null)),
+                    $queryBuilder->expr()->eq('document_format', $queryBuilder->createNamedParameter(''))
+                ),
```

Comment:
This comma (`,`) breaks TYPO3 8.7 ;-)
```
syntax error, unexpected ')'
```

Reply:
yes :-) thats the reason why TYPO3 is named TYPO3 :-D a simple syntax error. fixed.